data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,441 @@
+        ])
+        df_to_append['project_id'] = date_fields_df['table_catalog']
+        df_to_append['dataset_id'] = date_fields_df['table_schema']
+        df_to_append['table'] = date_fields_df['table_name']
+        df_to_append = df_to_append.drop_duplicates()
+
+        # Create new df to loop through date time fields
+        df_to_iterate = pd.DataFrame(
+            columns=['project_id', 'dataset_id', 'table', 'column'])
+        df_to_iterate['project_id'] = date_fields_df['table_catalog']
+        df_to_iterate['dataset_id'] = date_fields_df['table_schema']
+        df_to_iterate['table'] = date_fields_df['table_name']
+        df_to_iterate['column'] = date_fields_df['column_name']
+
+        # Remove person table and death table
+        df_to_append = df_to_append[~df_to_append.table.str.
+                                    contains('death', 'person')]
+        df_to_iterate = df_to_iterate[~df_to_iterate.table.str.
+                                      contains('death', 'person')]
+
+        # Filter through date columns and append to the appropriate column
+        for i, row in df_to_iterate.iterrows():
+            column = getattr(row, 'column')
+            table = getattr(row, 'table')
+            if 'start_date' in column:
+                df_to_append.loc[df_to_append.table == table,
+                                 'start_date_column'] = column
+            elif 'end_date' in column:
+                df_to_append.loc[df_to_append.table == table,
+                                 'end_date_column'] = column
+            else:
+                df_to_append.loc[df_to_append.table == table,
+                                 'date_column'] = column
+
+        date_fields_info_df = date_fields_info_df.append(df_to_append)
+
+    return date_fields_info_df
+
+
+def is_deid_dataset(dataset_id):
+    """
+    Determine if a dataset is deid using regex defined above
+    :param dataset_id: dataset_id to verify if it is deid
+
+    :return: boolean
+    """
+    return bool(re.match(DEID_REGEX, dataset_id))
+
+
+def get_research_id(project, dataset, pid):
+    """
+    For deid datasets, this function queries the _deid_map table in the associated combined dataset based on the release
+    regex prefix; to return the research_id for the specific pid passed
+
+    :param project: bq name of project
+    :param dataset: bq name of dataset
+    :param pid: person_id passed to receive associated research_id
+    :return: research_id or None if it does not exist for that person_id
+    """
+    # Get non deid dataset prefix
+    prefix = dataset.split('_')[0]
+    prefix = prefix.replace('R', '')
+
+    research_id_df = bq.query(
+        RESEARCH_ID_QUERY.format(project=project, prefix_regex=prefix, pid=pid))
+    if research_id_df.empty:
+        logging.info('no research_id associated with person_id: %s ' % pid)
+        return None
+    else:
+        return research_id_df.to_string()
```

Comment:
You could also write this as:
```
research_id_df.empty:
    logging.info(f'no research_id associated with person_id: {pid}')
    return ''   # or return None

return research_id_df.to_string()
```
The else statement here is unnecessary because it will execute the return even without the `else`.  Consider returning an empty string because it will act the same as `None` in conditional statements.